jujuclient/setmodels_test.go

```diff
@@ -0,0 +1,170 @@
+		"admin/another-model": s.assertModelStored(c, "admin/another-model", "test.model.uuid.2"),
+	}
+	s.assertStoreUpdated(c, expected)
+	s.assertStoreUpdated(c, expected)
+}
+
+func (s *ControllerModelsSuite) TestSetModelsDeleteOne(c *gc.C) {
+	s.assertControllerStored(c)
+	detailsToLeave := s.assertModelStored(c, "admin/new-model", "test.model.uuid")
+	before := map[string]jujuclient.ModelDetails{
+		"admin/new-model":     detailsToLeave,
+		"admin/another-model": s.assertModelStored(c, "admin/another-model", "test.model.uuid.2"),
+	}
+	after := map[string]jujuclient.ModelDetails{
+		"admin/new-model": detailsToLeave,
+	}
+
+	storedModels, err := s.store.AllModels(s.controllerName)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(storedModels, gc.DeepEquals, before)
+	s.assertStoreUpdated(c, after)
+}
+
+func (s *ControllerModelsSuite) TestSetModelsDeleteAll(c *gc.C) {
+	s.assertControllerStored(c)
+	before := map[string]jujuclient.ModelDetails{
+		"admin/new-model":     s.assertModelStored(c, "admin/new-model", "test.model.uuid"),
+		"admin/another-model": s.assertModelStored(c, "admin/another-model", "test.model.uuid.2"),
+	}
+	storedModels, err := s.store.AllModels(s.controllerName)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(storedModels, gc.DeepEquals, before)
+	s.assertStoreUpdated(c, nil)
+}
+
+func (s *ControllerModelsSuite) TestSetModelsAddUpdateDeleteCombination(c *gc.C) {
+	s.assertControllerStored(c)
+	detailsToUpdate := s.assertModelStored(c, "admin/update-model", "test.model.uuid.2")
+	before := map[string]jujuclient.ModelDetails{
+		"admin/delete-model": s.assertModelStored(c, "admin/delete-model", "test.model.uuid"),
+		"admin/update-model": detailsToUpdate,
+	}
+	after := map[string]jujuclient.ModelDetails{
+		"admin/new-model":    jujuclient.ModelDetails{"test.model.uuid"},
+		"admin/update-model": detailsToUpdate,
+	}
+
+	storedModels, err := s.store.AllModels(s.controllerName)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(storedModels, gc.DeepEquals, before)
+	s.assertStoreUpdated(c, after)
+}
+
+func (s *ControllerModelsSuite) TestSetModelsControllerIsolated(c *gc.C) {
+	s.assertControllerStored(c)
+	before := map[string]jujuclient.ModelDetails{
+		"admin/new-model": s.assertModelStored(c, "admin/new-model", "test.model.uuid"),
+	}
+
+	s.controller.ControllerUUID = "another.test.kontroller.uuid"
+	err := s.store.AddController("another-kontroller", s.controller)
+	c.Assert(err, jc.ErrorIsNil)
+	otherModels := map[string]jujuclient.ModelDetails{
+		"admin/foreign-model": jujuclient.ModelDetails{"test.foreign.model.uuid"},
+	}
+	err = s.store.SetModels("another-kontroller", otherModels)
+	c.Assert(err, jc.ErrorIsNil)
+
+	// initial controller un-touched
+	storedModels, err := s.store.AllModels(s.controllerName)
+	c.Assert(err, jc.ErrorIsNil)
+	c.Assert(storedModels, gc.DeepEquals, before)
+}
+
+func (s *ControllerModelsSuite) assertStoreUpdated(c *gc.C, models map[string]jujuclient.ModelDetails) {
```

Comment:
it's hard to follow the tests, because the method names don't do what I would think they do based on their names

s/assertStoreUpdated/assertSetModels/ please